ui/src/pages/Upload.tsx

```diff
@@ -155,8 +184,30 @@ export const Upload: React.FC = () => {
     });
   };
 
+  const handleDialogClose = () => {
+    setDialogOpenState(false);
+  };
+
+  const handleDialogAgree = () => {
+    handleDialogClose();
+  };
+
+  const handleDialogDisagree = () => {
+    handleDialogClose();
+  };
+
+  const storySubmitDialog = (result) => {
+    setLoading(false);
+    const resultMessage = JSON.parse(result).message;
+    if (resultMessage === "Story Added Successfully") {
+      setDialogOpenState(true);
+    } else {
+      setErrorOpen(true);
```

Comment:
We will need `setErrorOpen(true)` if the `.catch()` gets executed (promise gets rejected)